File: Math and Logic/Prime Number/Prime.cpp
#include <iostream>

using namespace std;

bool isPrime(int number)
{
    if (number == 1 || number == 0)
    {
        return false;
    }

    int limit = (int)(number / 2) + 1;
    for (int i = 2; i < limit; i++)
    {
        if (number % i == 0)
            return false;
    }
    return true;
}

int main()
{

    int number;
    cout << "Enter a number: ";
    cin >> number;

    string result = (isPrime(number) == true) ? "a prime." : "not a prime.";

    cout << "Entered number is " << result;
    return 0;
}

File: Math and Logic/Prime Number/Prime.py
def isPrime(number):
    if number == 1 or number == 0:
        return  False
        
    limit = (number // 2) + 1
    for i in range(2, limit):
        if number % i == 0:
            return False

    return True

if __name__ == "__main__":
    number = int(input("Enter a number: "))
    print("Entered number is", "a prime." if isPrime(number) == True else "not a prime.")